internal/services/oracle/oracle_autonomous_database_cross_region_disaster_recovery_resource.go

```diff
@@ -0,0 +1,469 @@
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			param := autonomousdatabases.AutonomousDatabase{
+				Name:     pointer.To(model.Name),
+				Location: location.Normalize(model.Location),
+				Tags:     pointer.To(model.Tags),
+				Properties: &autonomousdatabases.AutonomousDatabaseCrossRegionDisasterRecoveryProperties{
+					IsReplicateAutomaticBackups:    pointer.To(model.ReplicateAutomaticBackupsEnabled),
+					RemoteDisasterRecoveryType:     autonomousdatabases.DisasterRecoveryType(model.RemoteDisasterRecoveryType),
+					Source:                         autonomousdatabases.Source(model.Source),
+					SourceId:                       model.SourceId,
+					SourceOcid:                     pointer.To(model.SourceOcid),
+					SourceLocation:                 pointer.To(model.SourceLocation),
+					AdminPassword:                  pointer.To(model.AdminPassword),
+					BackupRetentionPeriodInDays:    pointer.To(model.BackupRetentionPeriodInDays),
+					CharacterSet:                   pointer.To(model.CharacterSet),
+					ComputeCount:                   pointer.To(model.ComputeCount),
+					ComputeModel:                   pointer.To(autonomousdatabases.ComputeModel(model.ComputeModel)),
+					CustomerContacts:               pointer.To(expandAdbsCustomerContacts25(model.CustomerContacts)),
+					DataBaseType:                   autonomousdatabases.DataBaseTypeCrossRegionDisasterRecovery,
+					DataStorageSizeInTbs:           pointer.To(model.DataStorageSizeInTbs),
+					DbWorkload:                     pointer.To(autonomousdatabases.WorkloadType(model.DbWorkload)),
+					DbVersion:                      pointer.To(model.DbVersion),
+					DisplayName:                    pointer.To(model.DisplayName),
+					IsAutoScalingEnabled:           pointer.To(model.AutoScalingEnabled),
+					IsAutoScalingForStorageEnabled: pointer.To(model.AutoScalingForStorageEnabled),
+					IsMtlsConnectionRequired:       pointer.To(model.MtlsConnectionRequired),
+					LicenseModel:                   pointer.To(autonomousdatabases.LicenseModel(model.LicenseModel)),
+					NcharacterSet:                  pointer.To(model.NationalCharacterSet),
+					SubnetId:                       pointer.To(model.SubnetId),
+					VnetId:                         pointer.To(model.VnetId),
+				},
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r AutonomousDatabaseCrossRegionDisasterRecoveryResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.AutonomousDatabases
+			id, err := autonomousdatabases.ParseAutonomousDatabaseID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %q: %+v", metadata.ResourceData.Id(), err)
```

Comment:
for Resource ID parsing error we just directly return the err without wrapping it

```suggestion
				return err
```